### ISSUE TITLE
Isolate RTP-specific code and audio encoding/decoding

### ISSUE THREAD

**Issue by gavv** [roc-project/roc#58]
After #48 and #57, it became clear what components rely on RTP-specific packet parameters. Such code now explicitly uses `Packet::rtp()`.

We can make most of those components protocol-independent and/or isolate RTP-specific parts.

**Comment by gavv** [comment]
After refactoring, RTP is used explicitly in the following components:

* audio::Packetizer
* audio::Depacketizer
* fec::Writer
* fec::Reader
* rtp::Composer
* rtp::Parser
* rtp::Validator
* rtp::PCMEncoder
* rtp::PCMDecoder

Now we have to move RTP-specific part from audio::Packetizer and audio::Depacketizer to rtp::PCMEncoder and rtp::PCMDecoder and we're done. We'll do it in #51.

**Comment by gavv** [comment]
It appeared that rtp::PCMEncoder and rtp::PCMDecoder are not the place where we should move the code that fill RTP header fields (seqnum/timestamp). There are two reasons of not doing this:

* we'll have to duplicate this code for every codec
* codecs will be tied to protocol: we won't be able to re-use the same codec with different protocols
* codecs will be tied to packets: we won't be able to use codecs for frames instead of packets, or e.g. to use multiple codecs to encode different parts of the same packet

Thus, this code should be either in Packetizer/Depacketizer on in a new abstraction layer between Packetizer/Depacketizer and codecs.

The later will allow us to make Packetizer/Depacketizer protocol-independent, but for now it's not clear what interface is needed for such a layer without knowing what specific protocols except RTP should it abstract. I think we can postpone it until we'll be implementing such a protocol.

So, the plan is:

* make codecs protocol-independent and move from roc_rtp to roc_audio;
* keep Packetizer and Depacketizer RTP-specific for now.

This will be covered by #211.
